src/silhouette.jl

```diff
@@ -101,12 +171,60 @@ function silhouettes(assignments::AbstractVector{<:Integer},
             #If both a[i] and b[i] are equal to 0 or Inf, silhouettes is defined as 0
             @inbounds sil[j] = a[j] < b[j] ? 1 - a[j]/b[j] :
                                a[j] > b[j] ? b[j]/a[j] - 1 :
-                               zero(eltype(r))
+                               zero(eltype(cluster_to_point))
         end
     end
     return sil
 end
 
+
+"""
+    silhouettes(assignments::AbstractVector, [counts,] dists) -> Vector{Float64}
+    silhouettes(clustering::ClusteringResult, dists) -> Vector{Float64}
+    silhouettes(x::AbstractMatrix, assignments::AbstractVector, pre::PrecomputeSilhouettes) -> Vector{Float64}
+    silhouettes(metric::Union{Metric, SemiMetric}, assignments::AbstractVector{<:Integer}, x::AbstractMatrix; 
+                nclusters=maximum(assignments), batch_size=size(x, 2), pre_calculate::Bool=true)
+
+Compute *silhouette* values for individual points w.r.t. given clustering.
+
+Returns the ``n``-length vector of silhouette values for each individual point.
+
+# Arguments
+ - `assignments::AbstractVector{Int}`: the vector of point assignments
+   (cluster indices)
+ - `counts::AbstractVector{Int}`: the optional vector of cluster sizes (how many
+   points assigned to each cluster; should match `assignments`)
+ - `clustering::ClusteringResult`: the output of some clustering method
+ - `dists::AbstractMatrix`: ``n×n`` matrix of pairwise distances between
+   the points
+ - `x::AbstractMatrix`: `d×n`` matrix of ``n`` data features of dimensionality ``d``
+ - `pre::PrecomputeSilhouettes`: precomputed vectors of cluster silhouettes.
+ - `metric::Union{Metric, SemiMetric}`: Distances metric object specifying which 
+   metric should be used to calculate the distances when given x.
+ - `nclusters` optional, how many clusters are associated with the assignments.
+ - `batch_size` optional, what batch size to use for distributed calculation, 
+    when available. Does not apply to direct use of `PrecomputeSilhouettes`.
+ - `pre_calculate` optional, control use of distributed algorithm (recommended by default).
+
+# References
+> Peter J. Rousseeuw (1987). *Silhouettes: a Graphical Aid to the
+> Interpretation and Validation of Cluster Analysis*. Computational and
+> Applied Mathematics. 20: 53–65.
+> Marco Gaido (2023). Distributed Silhouette Algorithm: Evaluating Clustering on Big Data 
+"""
+function silhouettes(assignments::AbstractVector{<:Integer},
+                     counts::AbstractVector{<:Integer},
+                     dists::AbstractMatrix{T}) where T<:Real
+
+    cluster_to_point = sil_aggregate_distances_normalized(assignments, counts, dists)
+    return silhouette_scores(cluster_to_point, assignments, counts)
+end
+
+function silhouettes(x::AbstractMatrix{T}, assignments::AbstractVector{<:Integer}, pre::PrecomputeSilhouettes) where T<:Real
```

Comment:
```suggestion
function silhouettes(pre::PrecomputeSilhouettes, assignments::AbstractVector{<:Integer}, points::AbstractMatrix{T}) where T<:Real
```
If it actually distances or point coordinates? In the former case the parameter should be called `dists`.

Reply:
Here it is actual coordinates.
Is there a reason for the change of places?
If there is, lets see how we can be consistent across all these places